zap/src/main/java/org/parosproxy/paros/extension/option/OptionsViewPanel.java

```diff
@@ -826,6 +828,10 @@ public void initParam(Object obj) {
                 .setSelected(options.getViewParam().isAllowAppIntegrationInContainers());
     }
 
+    private void setBreakPointRegistration(BreakPointRegistration registration) {
+        breakPointRegistration = registration;
+    }
+
```

Comment:
@thc202 I am not able to understand from where will this function will be called to set the breakPointRegistration interface. Sorry I am not good with design patterns, if you can tell which design pattern is being used here it could be helpful.

Reply:
what I am thinking is to access the OptionsViewPanel instance form the addon using reflection and then register it? I am not sure if this approach is correct or not

Reply:
oh okay got it. Thank you